mmdet/core/general_data/general_data.py

```diff
@@ -0,0 +1,498 @@
+
+          META INFORMATION
+        img_shape: (800, 1196, 3)
+        pad_shape: (800, 1216, 3)
+
+          DATA FIELDS
+        shape of det_labels: torch.Size([4])
+        shape of det_scores: torch.Size([4])
+
+        ) at 0x7f84acd10f90>
+        >>> resutls.det_scores
+        tensor([0.0100, 0.1000, 0.2000, 0.3000])
+        >>> results.det_labels
+        tensor([0, 1, 2, 3])
+        >>> results['det_labels']
+        tensor([0, 1, 2, 3])
+        >>> 'det_labels' in results
+        True
+        >>> results.img_shape
+        (800, 1196, 3)
+        >>> 'det_scores' in results
+        True
+        >>> del results.det_scores
+        >>> 'det_scores' in results
+        False
+        >>> det_labels = results.pop('det_labels', None)
+        >>> det_labels
+        tensor([0, 1, 2, 3])
+        >>> 'det_labels' in results
+        >>> False
+    """
+
+    def __init__(self, meta_info=None, data=None):
+
+        self._meta_info_fields = set()
+        self._data_fields = set()
+
+        if meta_info is not None:
+            self.set_meta_info(meta_info=meta_info)
+        if data is not None:
+            self.set_data(data)
+
+    def set_meta_info(self, meta_info=None):
+        """Add meta information.
+
+        Args:
+            meta_info (dict): A dict contains the meta information
+                of image. such as `img_shape`, `scale_factor`, etc.
+                Default: None.
+        """
+        assert isinstance(meta_info,
+                          dict), f'meta should be a `dict` but get {meta_info}'
+        meta = copy.deepcopy(meta_info)
+        for k, v in meta.items():
+            # should be consistent with original meta_info
+            if k in self._meta_info_fields:
+                ori_value = getattr(self, k)
+                if isinstance(ori_value, (torch.Tensor, np.ndarray)):
+                    if (ori_value == v).all():
+                        continue
+                    else:
+                        raise KeyError(
+                            f'img_meta_info {k} has been set as '
+                            f'{getattr(self, k)} before, which is immutable ')
+                elif ori_value == v:
+                    continue
+                else:
+                    raise KeyError(
+                        f'img_meta_info {k} has been set as '
+                        f'{getattr(self, k)} before, which is immutable ')
+            else:
+                self._meta_info_fields.add(k)
+                self.__dict__[k] = v
+
+    def set_data(self, data=None):
+        """Update a dict to `data_fields`.
+
+        Args:
+            data (dict): A dict contains annotations of image or
+                model predictions. Default: None.
+        """
+        assert isinstance(data,
+                          dict), f'meta should be a `dict` but get {data}'
+        for k, v in data.items():
+            self.__setattr__(k, v)
+
+    def new_results(self, meta_info=None, data=None):
```

Comment:
Since the class name is not `Results` now, should this function be named to `new_data` or just `new`?

Reply:
Good suggestion, I have renamed it to `new`